lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -43,6 +43,32 @@ void SubstraitDialect::initialize() {
 #include "substrait-mlir/Dialect/Substrait/IR/SubstraitOpInterfaces.cpp.inc"
 #include "substrait-mlir/Dialect/Substrait/IR/SubstraitTypeInterfaces.cpp.inc"
 
+//===----------------------------------------------------------------------===//
+// Custom Parser and Printer for Substrait
+//===----------------------------------------------------------------------===//
+
+bool parseCountAsAll(OpAsmParser &parser, IntegerAttr &count) {
+  if (!parser.parseOptionalKeyword("all")) {
+    count = parser.getBuilder().getI64IntegerAttr(-1);
+  } else {
+    int64_t result;
+    if (!parser.parseInteger(result)) {
+      count = parser.getBuilder().getI64IntegerAttr(result);
+    } else {
+      return true;
+    }
+  }
+  return false;
+}
+
+void printCountAsAll(OpAsmPrinter &printer, Operation *op, IntegerAttr count) {
+  if (count.getInt() == -1) {
+    printer << "all";
+    return;
+  } else {
+    printer << count.getValue();
+  }
```

Comment:
Similar here: if you want to keep the `else` branch, then also `return` from it. But I think this is clearer:
```suggestion
  }
  
  // Normal integer.
  printer << count.getValue();
```